pkg/kubelet/kubelet_pods.go

```diff
@@ -1741,46 +1741,79 @@ func getPhase(pod *v1.Pod, info []v1.ContainerStatus, podIsTerminal bool) v1.Pod
 	}
 }
 
-func deleteCustomResourceFromResourceRequirements(target *v1.ResourceRequirements) {
-	for resource := range target.Limits {
-		if resource != v1.ResourceCPU && resource != v1.ResourceMemory && resource != v1.ResourceEphemeralStorage {
-			delete(target.Limits, resource)
+func (kl *Kubelet) determinePodResizeStatus(allocatedPod *v1.Pod, podStatus *kubecontainer.PodStatus, podIsTerminal bool) v1.PodResizeStatus {
+	if kubetypes.IsStaticPod(allocatedPod) {
+		return ""
+	}
+
+	// If pod is terminal, clear the resize status.
+	if podIsTerminal {
+		if err := kl.statusManager.SetPodResizeStatus(allocatedPod.UID, ""); err != nil {
+			klog.ErrorS(err, "SetPodResizeStatus failed for terminal pod", "pod", format.Pod(allocatedPod))
 		}
+		return ""
 	}
-	for resource := range target.Requests {
-		if resource != v1.ResourceCPU && resource != v1.ResourceMemory && resource != v1.ResourceEphemeralStorage {
-			delete(target.Requests, resource)
+
+	resizeStatus, _ := kl.statusManager.GetPodResizeStatus(string(allocatedPod.UID))
+	// If the resize was in-progress and the actual resources match the allocated resources, mark
```

Comment:
If there are multiple resize requests, the 1st is in-progress while the 2nd one is yet to be admitted by kubelet, this would clear the status, so users see:
 - desired != allocated
 - allocated == actual
 - resize status: empty

On the other hand, if the resize status is in "Proposed" (and not clobbered by kubelet), users see
 - desired != allocated
 - allocated == actual
 - resize status: proposed

The "proposed" status is not useful at all. Users should simply check the desired/allocated/actual status to understand if a request has been actuated(?).

Reply:
Trying to figure out how users should interpret the resize status:
 1. Proposed: not meaningful. Check the resource values to understand more. (potentially desired != allocated)
 2. InProgress: some resizing request is in progress (can also be derived from allocated != actual)
 3. Deferred: kubelet's trying to see if it can admit a resized pod (can also be derived from desired != allocated)
 4. Infeasible: kubelet cannot admit a version of the resized pod
 5. "" (empty): kubelet's not currently working any resize request (even if there might be a new desired value)

So a pod can be in a more than one status (e.g., in-progress and deferred) theoretically. I wonder if these would be better surfaced as pod conditions instead -- some of them can be independent and we don't have to check on previous values.

Reply:
Discussed over chat. I agree with your analysis, but for v1.32 proceed with it as-is and document the issues, and look into improvements in v1.33. We may not be able to fully resolve this without something like the synchronous admission proposed in https://docs.google.com/document/d/1O9rDeYhGWMbtg-DQmhqKvNsQSHWJSwvSEeNpXTKMvZg/edit?tab=t.0#heading=h.6ggs708p6vvu

Reply:
Yep, we'll have to work around the limitations of the current design for 1.32. Feel free to resolve this comment. 

cc @thockin and @dchen1107 FYI.